x-pack/legacy/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -262,6 +268,36 @@ export function getWaterfall(
     entryWaterfallTransaction
   );
 
+  function isAncestor(
+    waterfallItemsList: IWaterfallItem[],
+    ancestorTransactionId: string,
+    currentItem: IWaterfallError | IWaterfallItem
+  ): boolean {
+    if (!currentItem.parentId) {
+      return false;
+    }
+    if (currentItem.parentId === ancestorTransactionId) {
+      return true;
+    }
+
+    const parentItem: IWaterfallItem | undefined = waterfallItemsList.find(
+      ({ id }) => id === currentItem.parentId
+    );
+
+    if (parentItem) {
+      return isAncestor(waterfallItemsList, ancestorTransactionId, parentItem);
+    }
+    return false;
+  }
+
+  const errorItems: IWaterfallError[] = trace.errorDocs
+    .map(errorDoc =>
+      getErrorItem(errorDoc as APMError, items, entryWaterfallTransaction)
+    )
+    .filter(errorItem => {
+      return isAncestor(items, entryTransactionId, errorItem);
```

Comment:
Not 100% sure if I understand what this is doing. Is it filtering out any errors that are not in the trace?

Reply:
it makes sure to only return errors for which the entryTransaction from the URL params is an ancestor. If they focus on a particular transaction in the waterfall, we don't want to show all the errors of the trace, just those that apply to the currently rendered waterfall.